app/tests/core_tests/test_fields.py

```diff
@@ -0,0 +1,24 @@
+from contextlib import nullcontext
+
+import pytest
+from django.core.exceptions import ValidationError
+
+from grandchallenge.core.fields import RegexField
+
+
+@pytest.mark.parametrize(
+    "value, error",
+    (
+        (
+            r"[abc]",
+            nullcontext(),
+        ),
+        (
+            r"[abc",  # Missing closing bracket
+            pytest.raises(ValidationError),
+        ),
+    ),
+)
+def test_regex_field_validation(value, error):
```

Comment:
Would be good to check that complex patterns get compiled correctly and the correct regex is returned, e.g. `r"/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/"`.

Also better to go through Django's way of validating fields rather than calling the private validation method here and trusting that it's all hooked up correctly. You can create a temporary models to do this.

Reply:
Using the a temp model for the field test turned out to be relatively simple:

```Python
class ModelWithRegex(models.Model):
    regex = RegexField()


@pytest.mark.parametrize(
    "value, error",
    (
        (
            r"[abc]",
            nullcontext(),
        ),
        (
            r"[abc",  # Missing closing bracket
            pytest.raises(ValidationError),
        ),
        (
            r"/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/",
            nullcontext(),
        ),
    ),
)
def test_regex_field_validation(value, error):
    model = ModelWithRegex(regex=value)
    with error:
        model.full_clean()
```

However, I am not too sure how to test if the pattern survives DB entry intact without actually using the DB. Do you have a suggestion?

Reply:
Yes, the only way to test that would be to save to the db and read back. You are already using the field in a model, so could do that there.